vita3k/app/src/app_init.cpp

```diff
@@ -101,11 +102,82 @@ void update_viewport(EmuEnvState &state) {
     }
 }
 
+void init_paths(Root &root_paths) {
+    auto base_path = SDL_GetBasePath();
+    auto pref_path = SDL_GetPrefPath(org_name, app_name);
+    root_paths.set_base_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_pref_path(string_utils::utf_to_wide(pref_path));
+    root_paths.set_log_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_config_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_shared_path(string_utils::utf_to_wide(base_path));
+    root_paths.set_cache_path(string_utils::utf_to_wide(base_path));
+    SDL_free(base_path);
+    SDL_free(pref_path);
+
+#if defined(__linux__) && !defined(__ANDROID__) && !defined(__APPLE__)
+    // XDG Data Dirs.
+    auto env_home_dir = getenv("HOME");
```

Comment:
this will always be set according to POSIX specification (https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap08.html)
> HOME
The system shall initialize this variable at the time of login to be a pathname of the user's home directory.

Reply:
The user, or some script, can still undefine it. Which will cause a NULL de-reference if we just use the given pointer without checking it first.

Should I remove the check anyway?

Reply:
yeah, even if a script removes it that would be asking for problems, theres no reason to remove a posix standard env var, also i have never seen a script do that lol

Reply:
Note: Making this comment to give a more detail as to why this change was reverted in current versions of this PR.

OK, after removing the check and replacing $HOME with '~/', Vita3K started to create a '~' directory in the ${PWD}. (In my case, in the build directory of the source tree.) It seems that boost isn't resolving '~' to the user's actual home directory.

Worse, if the created '~' directory is on a different filesystem than the user's actual home directory, attempting to call copy_file() fails with a boost filesystem exception. Apparently, it's invoking behavior similar to a file _rename_ instead of a file copy. Which seems really wrong. (Perhaps boost's copy file implementation recognizes the '~' and tries a rename given the destination path is also HOME just spell out? But if that's the case why does it create the '~' directory in the first place????)

Regardless, I've reverted this change in the current commits. This needs to be consistent or Vita3K is going to generate random errors all over the place. If it's going to choke on '~/', may as well just use the resolved $HOME environment variable to avoid the disambiguation to begin with.